runtime/src/main/java/org/corfudb/protocols/CorfuProtocolCommon.java

```diff
@@ -0,0 +1,209 @@
+    public static final byte PROTO_CORFU_RESPONSE_MSG_MARK = 0x3;
+
+    /**
+     * Returns the Protobuf representation of a UUID.
+     *
+     * @param uuid   the desired (Java) UUID
+     */
+    public static UuidMsg getUuidMsg(UUID uuid) {
+        return UuidMsg.newBuilder()
+                .setLsb(uuid.getLeastSignificantBits())
+                .setMsb(uuid.getMostSignificantBits())
+                .build();
+    }
+
+    /**
+     * Returns a UUID object from its Protobuf representation.
+     *
+     * @param msg   the Protobuf UUID message
+     */
+    public static UUID getUUID(UuidMsg msg) {
+        return new UUID(msg.getMsb(), msg.getLsb());
+    }
+
+    /**
+     * Returns the Protobuf representation of a Layout.
+     *
+     * @param layout   the desired (Java) Layout
+     */
+    public static LayoutMsg getLayoutMsg(Layout layout) {
+        return LayoutMsg.newBuilder()
+                .setLayoutJson(layout.asJSONString())
+                .build();
+    }
+
+    /**
+     * Returns a Layout object from its Protobuf representation.
+     *
+     * @param msg   the Protobuf Layout message
+     */
+    public static Layout getLayout(LayoutMsg msg) {
+        return Layout.fromJSONString(msg.getLayoutJson());
+    }
+
+    /**
+     * Returns the Protobuf representation of a Token, given
+     * its epoch and sequence.
+     *
+     * @param epoch      the epoch of the token
+     * @param sequence   the sequencer number of the token
+     */
+    public static TokenMsg getTokenMsg(long epoch, long sequence) {
+        return TokenMsg.newBuilder()
+                .setEpoch(epoch)
+                .setSequence(sequence)
+                .build();
+    }
+
+    /**
+     * Returns the Protobuf representation of a Token.
+     *
+     * @param token   the desired (Java) Token
+     */
+    public static TokenMsg getTokenMsg(Token token) {
+        return getTokenMsg(token.getEpoch(), token.getSequence());
+    }
+
+    /**
+     * Returns the Protobuf representation of a SequencerMetrics object.
+     *
+     * @param metrics   the desired (Java) SequencerMetrics
+     */
+    public static SequencerMetricsMsg getSequencerMetricsMsg(SequencerMetrics metrics) {
+        return SequencerMetricsMsg.newBuilder()
+                .setSequencerStatus(sequencerStatusTypeMap.get(metrics.getSequencerStatus()))
+                .build();
+    }
+
+    /**
+     * Returns a SequencerMetrics object from its Protobuf representation.
+     *
+     * @param msg   the Protobuf SequencerMetrics message
+     */
+    public static SequencerMetrics getSequencerMetrics(SequencerMetricsMsg msg) {
+        switch(msg.getSequencerStatus()) {
+            case READY: return SequencerMetrics.READY;
+            case NOT_READY: return SequencerMetrics.NOT_READY;
+            case UNKNOWN: return SequencerMetrics.UNKNOWN;
+            default:
+        }
+
+        //TODO: Revisit for rolling-upgrades?
+        return null;
```

Comment:
You might want to throw an `InvalidStateException` insated in the default block.